src/NHibernate/Impl/Printer.cs

```diff
@@ -72,11 +71,16 @@ public string ToString(IType[] types, object[] values)
 
 		public string ToString(IDictionary<string, TypedValue> namedTypedValues)
 		{
-			IDictionary<string, string> result = new Dictionary<string, string>(namedTypedValues.Count);
+			return ToString(namedTypedValues.AsEnumerable());
+		}
+
+		internal string ToString(IEnumerable<KeyValuePair<string, TypedValue>> namedTypedValues)
+		{
+			IDictionary<string, string> result = new Dictionary<string, string>(namedTypedValues.Count());
```

Comment:
Not happy with this line at all.

Reply:
It does the same than previous code though.

Reply:
I have added a fixup with another implementation, leaving the old method (still used at other places) untouched.